### PR TITLE
Update onboarding checklist template

### PR DESCRIPTION
**Pull request by tobie**
Follows https://github.com/openjs-foundation/cross-project-council/pull/697.

### CODE REVIEW

**Review by ljharb — APPROVED** [review]

**Review by eemeli — APPROVED** [review]
This seems like just the sort of thing that ought to be automated with a GitHub Action.

File: .github/ISSUE_TEMPLATE/project-onboarding-checklist-template.md [hunk 1]
```diff
@@ -9,26 +9,28 @@ assignees: ''
 
 ## Project Onboarding Checklist - ${Project Name}
 
-<!--Copied from ../../PROJECT_PROGRESSION.md on 2019-10-31 -->
+<!--Copied from ../../PROJECT_PROGRESSION.md on 2020-01-05 -->
```
[the remark this answers is not in the document]
Comment (eemeli) [reply .github/ISSUE_TEMPLATE/project-onboarding-checklist-template.md:12]: Also, it's not 2020 anymore! 🎉

**Comment by tobie** [comment]
> This seems like just the sort of thing that ought to be automated with a GitHub Action.

Oh, I hadn't thought of that. Smart!

File: .github/ISSUE_TEMPLATE/project-onboarding-checklist-template.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tobie) [reply .github/ISSUE_TEMPLATE/project-onboarding-checklist-template.md]: > Also, it's not 2020 anymore! 🎉

HOW CAN YOU BE SO SURE?

File: .github/ISSUE_TEMPLATE/project-onboarding-checklist-template.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ assignees: ''
 
 ## Project Onboarding Checklist - ${Project Name}
 
-<!--Copied from ../../PROJECT_PROGRESSION.md on 2020-01-05 -->
+<!--Copied from https://github.com/openjs-foundation/cross-project-council/blob/master/PROJECT_PROGRESSION.md on 2021-01-05 -->
```
Comment (ljharb) [inline .github/ISSUE_TEMPLATE/project-onboarding-checklist-template.md:12]:
```suggestion
<!--Copied from https://github.com/openjs-foundation/cross-project-council/blob/HEAD/PROJECT_PROGRESSION.md on 2021-01-05 -->
```

**Review by mhdawson — APPROVED** [review]
LGTM

**Pull request merged**